docs/guides/real-time/chat/react.md

```diff
@@ -87,177 +108,97 @@ const messageSubmit = (event) => {
-
-```js
-const [formValue, setFormValue] = useState({ email: '', password: '' });
-```
-
-Set up a `handleLoginChange` method that updates the value of the login input field as the user types.
-
-```js
-const handleLoginChange = (event) => {
-	setFormValue({ ...formValue, [event.target.name]: event.target.value });
-};
-```
-
-Then, connect these values to the form input fields:
-
-```jsx
-<form onSubmit={loginSubmit}>
-	<label htmlFor="email">Email</label>
-	<input type="email" id="email" /> // [!code --]
-	<input type="email" id="email" name="email" value={formValue.email} onChange={handleLoginChange} /> // [!code ++]
-	<label htmlFor="password">Password</label>
-	<input type="password" id="password" /> // [!code --]
-	<input type="password" id="password" name="password" value={formValue.password} onChange={handleLoginChange} /> // [!code ++]
-	<button type="submit">Submit</button>
-</form>
-```
-
-Within the `loginSubmit` method, create a new WebSocket, which will immediately attempt connection:
+Now, extract the `email` and `password` values from the login form.
 
 ```js
-const loginSubmit = (event) => {
-	connectionRef.current = new WebSocket(url); // [!code ++]
-};
+ const loginSubmit = (event) => {
+    event.preventDefault();
+    const email = event.target.elements.email.value; // [!code ++]
+    const password = event.target.elements.password.value; // [!code ++]
+  };
 ```
 
-On connection, you must [send an authentication message before the timeout](/guides/real-time/authentication). Add an
-event handler for the connection's `open` event:
+Once the client is authenticated, immediately create a WebSocket connection:
 
 ```js
-const loginSubmit = (event) => {
-	connectionRef.current = new WebSocket(url);
-	connectionRef.current.addEventListener('open', authenticate(formValue)); // [!code ++]
-};
+client.connect();
 ```
 
-Then, create a new `authenticate` method:
+## Subscribe To Messages
 
-```js
-const authenticate = (opts) => {
-	const { email, password } = opts;
-	connectionRef.current.send(JSON.stringify({ type: 'auth', email, password }));
-};
-```
-
-### Subscribe to Messages
-
-In a WebSocket connection, all data sent from the server will trigger the connection’s `message` event. Inside
-`loginSubmit`, add an event handler:
+As soon as you have successfully authenticated, a message will be sent. When this happens, within `useEffect`, subscribe
+to updates on the `Messages` collection.
 
 ```js
-const loginSubmit = (event) => {
-	connectionRef.current = new WebSocket(url);
-	connectionRef.current.addEventListener('open', authenticate(formValue));
-	connectionRef.current.addEventListener('message', (message) => receiveMessage(message)); // [!code ++]
-};
+useEffect(() => {
+  const cleanup = client.onWebSocket('message', function (data) {
+    if (data.type == 'auth' && data.status == 'ok') {
+      subscribe('create');
+    }
+    console.log(message);
```

Comment:
Was this log left here intentionally for debugging purposes?

Reply:
Yes when building the project 😅 but I'll take it out